### PR TITLE
Previewing destructured variables

### PR DESCRIPTION
**Pull request by marcogallo566**
Fixes Issue: #5706

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

### Summary of Changes

* Now hovering on ' { variable } = another; ' show the value

Here's the Debugger's Testing doc
https://docs.google.com/document/d/1oBMRxV8A2ag2t22YsQOxTdEv0mXKzIg0tggJjRkU1S0/edit#.
Feel free to improve it!

### Screenshots
1f6753af-a3b4-4169-a325-5708b2ba54ac

### CODE REVIEW

File: src/workers/parser/getSymbols.js [hunk 1]
```diff
@@ -256,14 +256,23 @@ function extractSymbol(path: SimplePath, symbols) {
...
       return;
     }
-    symbols.identifiers.push({
-      name: node.name,
-      expression: node.name,
-      location: { start, end }
-    });
+    if (properties != undefined && t.objectPattern(properties)) {
+      symbols.identifiers.push({
+        name: node.properties[0].value.name,
```
[the remark this answers is not in the document]
Comment (marcogallo566) [reply src/workers/parser/getSymbols.js]: I'll change the code to make it work. Thank you for showing me the error!

**Comment by jasonLaster** [comment]
We just landed #5693
I recommend pulling down master and rebasing it

Also, this might fix some of the cases too...

**Comment by marcogallo566** [comment]
Now it work with 2 variables:

93fd889e-fd0d-4688-80fc-e90dc1d5a2fb

File: src/workers/parser/getSymbols.js [hunk 2]
```diff
@@ -255,15 +255,27 @@ function extractSymbol(path: SimplePath, symbols) {
...
+    const { properties } = node;
     if (t.isArrayPattern(node)) {
       return;
     }
-    symbols.identifiers.push({
-      name: node.name,
-      expression: node.name,
-      location: { start, end }
-    });
+    if (properties != undefined && t.objectPattern(properties)) {
```
Comment (jasonLaster) [inline src/workers/parser/getSymbols.js]: NIT: this is clearer
```js
if (properties && t.objectPattern(properties)) {}
 ```
 
 is better if that works
 ```js
if (t.objectPattern(properties)) {}
 ```

File: src/workers/parser/getSymbols.js [hunk 3]
```diff
@@ -255,15 +255,27 @@ function extractSymbol(path: SimplePath, symbols) {
...
+        });
+      });
+    } else {
+      const { start, end } = path.node.loc;
+      symbols.identifiers.push({
+        name: node.name,
+        expression: node.name,
+        location: { start, end }
+      });
+    }
```
Comment (jasonLaster) [inline src/workers/parser/getSymbols.js]: it'd be nice to consolidate this and do something like:

```js
const properties = t.isObjectPattern(node.properties) ? node.properties: [path.node];
properties.forEach((property) => {
  const { start, end } = property.loc;
  // ...
})
```

File: src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap [hunk 4]
```diff
@@ -387,10 +387,12 @@ comments:
 
 
 identifiers:
-[(1, 6), (1, 35)]
+[(1, 8), (1, 9)]  b b
+[(1, 11), (1, 16)]  resty resty
 [(1, 21), (1, 28)]  compute compute
 [(1, 29), (1, 34)]  stuff stuff
-[(2, 6), (2, 33)]
+[(2, 8), (2, 16)]  f f
```
Comment (jasonLaster) [inline src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap]: nice!

File: src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap [hunk 5]
```diff
@@ -405,12 +407,13 @@ identifiers:
 [(8, 10), (8, 13)]  log log
 [(8, 14), (8, 19)]  index index
 [(8, 21), (8, 28)]  element element
-[(11, 6), (11, 42)]
+[(11, 8), (11, 18)]
+[(11, 20), (11, 29)]
```
Comment (jasonLaster) [inline src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap]: is it okay that these are empty

Comment (marcogallo566) [reply src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap]: Yes, I think because I'm solving a little issue, after this will be right.

Comment (marcogallo566) [reply src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap]: I understood these empty lines: when I hover on an expression like this `... = { a: 3 }` I get this error `TypeError: undefined has no properties`. This issue there was also before these changes, but I don't understand why before there was only one line.

File: src/workers/parser/getSymbols.js (hunk 3 above)
Comment (marcogallo566) [reply src/workers/parser/getSymbols.js]: A draft version of the changes:
````
  if (t.isVariableDeclarator(path)) {
    const node = path.node.id;
    const { properties } = node;
    const hasProperties = (properties && t.objectPattern(properties));
    if (t.isArrayPattern(node)) {
      return;
    }
    const listProperties = hasProperties ? node.properties : [path.node];
      listProperties.forEach(function(property) {
      const { start, end } = property.loc;
      symbols.identifiers.push({
        name: hasProperties ? property.value.name : property.name,
        expression: hasProperties ? property.value.name : property.name,
        location: { start, end }
      });
    });
...
````
But these changes 'amplify' the `TypeError: undefined has no properties` issue:

With changes | Without changes
------------ | -------------
b177f9ea-ce85-4c3d-b80f-b3d7d2ed141a | 96137cca-a833-4901-b324-10da91027124

This issue doesn't appear when I'm hovering `collection` in the same line.

Comment (bomsy) [reply src/workers/parser/getSymbols.js]: @marcogallo566 I would recommend you do a rebase first, there are some changes currently on master  that will fix the `undefined has no properties` issue and other things.

File: src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap (hunk 5 above)
Comment (bomsy) [reply src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap]: @marcogallo566 I would recommend you do a rebase first, there are changes on master that fixes this issue.

File: src/workers/parser/getSymbols.js (hunk 3 above)
Comment (marcogallo566) [reply src/workers/parser/getSymbols.js]: I did it, but probably I did it wrong. I'll try again this evening, thank you!

**Comment by darkwing** [comment]
Rebased this for you @marcogallo566 .  Going to do more manual 
testing this morning.

**Comment by marcogallo566** [comment]
@darkwing Thank you so much!

**Comment by darkwing** [comment]
Hey @marcogallo566 !  I attempted some code cleanup -- take a look and make sure everything you wanted was still there.  Thanks! :)

**Comment by darkwing** [comment]
Apologies for all the traffic on this -- refactoring the current var names was tripping me up in crazy ways :D

**Pull request merged**